### PR TITLE
feat: add values to analytics allow lists for Pre Apps

### PR DESCRIPTION
**Pull request by jessicamcinchak**
**Changes:**
- Adds 3x new values to analytics `ALLOW_LISTS` for Pre App services 
- Updates `analytics_summary` & `submission_services_summary` views 
  - Notably remembering to also include `GRANT SELECT ON "public"."{view_name}" TO metabase_read_only;` so that Metabase _should_ smoothly sync/pick up new columns :crossed_fingers:

### CODE REVIEW

File: api.planx.uk/modules/webhooks/service/analyzeSessions/operations.ts [hunk 1]
```diff
@@ -6,13 +6,18 @@ import type { Operation } from "../sanitiseApplicationData/types.js";
...
+ *  `analytics_summary` & `submission_services_summary` views to extract the value into its own column
+ * 
+ * Please also ensure your migration ends with `GRANT SELECT ON public.{VIEW_NAME} TO metabase_read_only`
+ *  so that Metabase picks up the new columns
  */
-
 const ALLOW_LIST = [
   "application.declaration.connection",
+  "application.information.harmful",
+  "application.information.sensitive",
```
Comment (jessicamcinchak) [inline api.planx.uk/modules/webhooks/service/analyzeSessions/operations.ts:20]: These are both yes/no questions - see here https://editor.planx.uk/doncaster/pre-application-advice

They additionally have extra "description" & "reason" free text inputs each, but those seem less chart-able & potentially sensitve info so I'm not including here - is that okay @zz-hh-aa ? Intention of analytics "allow lists" is to track generic, non-personally-identifiable stuff only in aggregate over all time. Doncaster of course will still get all answers with their submission data !

Screenshot from 2024-10-07 16-33-54

**Comment by jessicamcinchak** [comment]
[Expose `application.information.sensitive.description` in Metabase for pre-app dashboard](https://trello.com/c/vrdQnCh4/3044-expose-applicationinformationsensitivedescription-in-metabase-for-pre-app-dashboard)

**Comment by jessicamcinchak** [comment]
[Expose `application.information.harmful` in Metabase for pre-app dashboard](https://trello.com/c/2W4Mdr6O/3043-expose-applicationinformationharmful-in-metabase-for-pre-app-dashboard)

**Comment by jessicamcinchak** [comment]
[Expose `service.type` in Metabase for pre-app dashboard](https://trello.com/c/HXTSC1JM/3042-expose-servicetype-in-metabase-for-pre-app-dashboard)

File: api.planx.uk/modules/webhooks/service/analyzeSessions/operations.ts (hunk 1 above)
Comment (zz-hh-aa) [reply api.planx.uk/modules/webhooks/service/analyzeSessions/operations.ts:20]: Yes I think that'll do just fine! Thanks Jess :)

File: editor.planx.uk/src/pages/FlowEditor/lib/analytics/provider.tsx [hunk 2]
```diff
@@ -33,11 +33,19 @@ import {
...
- * If appending to ALLOW_LIST please also update the `analytics_summary` view to
- * extract it into it's own column.
+ * ALLOW_LIST should stay in sync with
+ *   api.planx.uk/modules/webhooks/service/analyzeSessions/operations.ts
+ *
+ * If appending values to ALLOW_LIST please also update the
+ *  `analytics_summary` & `submission_services_summary` views to extract the value into its own column
+ *
+ * Please also ensure your migration ends with `GRANT SELECT ON public.{VIEW_NAME} TO metabase_read_only`
+ *  so that Metabase picks up the new columns
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/lib/analytics/provider.tsx:43]: nit(non-blocking): Maybe these comments point towards the fact we might need a simple doc file explaining how to add values to `ALLOW_LIST`?

**Review by DafyddLlyr — APPROVED** [review]

**Pull request merged**